### PR TITLE
fix(eslint-plugin)!: remove "isTypescriptFile".

### LINKED ISSUE #524: Make typescript-eslint rules compatible with .vue files
My team and I are writing a Vue app using typescript. We have started making extensive use of [vue single file components](https://vuejs.org/v2/guide/single-file-components.html). The single file components end with a `.vue` file extension. Inside the components there are three possible sections:

* `<template></template>`
* `<script lang="ts"></script>`
* `<style></style>

The problem we are seeing is that some of the eslint plugin rules (`explicit-function-return-types` and `explicit-member-accessibility`) use a function in utils called `isTypescriptFile`. This check ensures that the rules are run only on `.ts` or `.tsx` files. This means that we are not getting some of our linting rules run in our `.vue` files. 

I can think of a couple good ways to fix this and would like some feedback on which is preferred :)
1. change the `isTypescriptFile` check to not only check the .`tsx?` extension but also check for `.vue` files which include a `<script lang="ts">` tag. 
2. Make the linter configurable on what files it should include as typescript files (run time variables, env variables, config file, etc). These variables would then replace the static file extension checker logic in `isTypescriptFile`. 

**Repro**

* Create a repo with a `.eslintrc.js` file. Which includes `@typescript-eslint` in its plugins and  '@typescript-eslint/explicit-function-return-type': 'error'` in its rules
* create a `.vue` file with `<script lang="ts"></script>`
* In that script tag create a function with no return type
* Linter will not catch the error

```JSON
{
  "rules": {
    "typescript/<rule>": ["<setting>"]
  }
}
```

```TS
// your repro code case
```

**Expected Result**
Linting of `.vue` files for all typescript-eslint rules. 

**Actual Result**
`explicit-function-return-types` and `explicit-member-accessibility` are not linting in `.vue` files. 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.9.0` |
| `@typescript-eslint/parser`        | `1.9.0` |
| `TypeScript`                       | `3.3.4` |
| `ESLint`                           | `5.15.3` |
| `node`                             | `11.9.0` |
| `npm`                              | `6.9.1` |

### PR DESCRIPTION
**Pull request by sosukesuzuki**
BREAKING CHANGE

Resolve #524 

Remove isTypescriptFile to make some typescript-eslint rules compate with files are not .ts or .tsx(e.g. .vue).

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
@bradzacher Sorry, I considered to pass the CodeCov, but I did not know what to do. Should probably I add some tests?

File: packages/eslint-plugin/src/rules/explicit-member-accessibility.ts [hunk 1]
```diff
@@ -119,30 +119,28 @@ export default util.createRule<Options, MessageIds>({
         return;
       }
 
-      if (util.isTypeScriptFile(context.getFilename())) {
-        // const methodName = util.getNameFromPropertyName(methodDefinition.key);
-        const methodName = util.getNameFromClassMember(
+      // const methodName = util.getNameFromPropertyName(methodDefinition.key);
```
Comment (JamesHenry) [inline packages/eslint-plugin/src/rules/explicit-member-accessibility.ts]: Please remove the commented out code

Comment (JamesHenry) [reply packages/eslint-plugin/src/rules/explicit-member-accessibility.ts]: Just seen that you didn't add this, it was already there, but if you wouldn't mind cleaning it up anyway that would be great!

**Comment by JamesHenry** [comment]
Just the one update, then we can merge this. Thanks @sosukesuzuki!

File: packages/eslint-plugin/src/rules/explicit-member-accessibility.ts (hunk 1 above)
Comment (sosukesuzuki) [reply packages/eslint-plugin/src/rules/explicit-member-accessibility.ts]: @JamesHenry Fix from 601159d!

**Pull request merged**